src/WebJobs.Script/SlidingWindow.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class SlidingWindow<TItem>
+    {
+        private readonly TimeSpan _window;
+        private List<Event> _events = new List<Event>();
+        private object _syncLock = new object();
+
+        public SlidingWindow(TimeSpan window)
+        {
+            _window = window;
+        }
+
+        public IEnumerable<TItem> GetEvents(Func<TItem, bool> filter)
+        {
+            lock (_syncLock)
+            {
+                // remove expired events
+                _events = _events.Where(p => (DateTime.Now - p.TimeStamp) < _window).ToList();
```

Comment:
Why not _events.RemoveAll(p => (DateTime.Now - p.TimeStamp) < _window)? Is there some advantage to constructing a new list here?

Reply:
I was unaware of the RemoveAll method - I've changed to use that :)